### ISSUE TITLE
service panic because of race conditions 

### ISSUE THREAD

**Issue by rvdwijngaard** [grafana/grafana#13674]
Grafana v5.3.0 (commit: c613a31)
Datasource: Cloudwatch 
OS : linux 

I have upgraded my server to the latest version. The service now panics with a random frequency. I cannot find a pattern yet. In the logs I get the following stack trace: 
```
fatal error: concurrent map writes
goroutine 12697 [running]:
runtime.throw(0x1523ebe, 0x15)
/usr/local/go/src/runtime/panic.go:608 +0x72 fp=0xc0009c7ea0 sp=0xc0009c7e70 pc=0x4353a2
runtime.mapassign_faststr(0x12ec080, 0xc00134c7b0, 0xc00121f0c0, 0x19, 0x0)
/usr/local/go/src/runtime/map_faststr.go:275 +0x3bf fp=0xc0009c7f08 sp=0xc0009c7ea0 pc=0x41b93f
github.com/grafana/grafana/pkg/tsdb/cloudwatch.(*CloudWatchExecutor).executeTimeSeriesQuery.func1(0x8, 0x1590fa0)
/go/src/github.com/grafana/grafana/pkg/tsdb/cloudwatch/cloudwatch.go:138 +0x1a2 fp=0xc0009c7f88 sp=0xc0009c7f08 pc=0xe8c272
github.com/grafana/grafana/vendor/golang.org/x/sync/errgroup.(*Group).Go.func1(0xc00134c810, 0xc001356380)
/go/src/github.com/grafana/grafana/vendor/golang.org/x/sync/errgroup/errgroup.go:58 +0x57 fp=0xc0009c7fd0 sp=0xc0009c7f88 pc=0xb2f347
runtime.goexit()
/usr/local/go/src/runtime/asm_amd64.s:1333 +0x1 fp=0xc0009c7fd8 sp=0xc0009c7fd0 pc=0x465221
created by github.com/grafana/grafana/vendor/golang.org/x/sync/errgroup.(*Group).Go
/go/src/github.com/grafana/grafana/vendor/golang.org/x/sync/errgroup/errgroup.go:55 +0x66
```

**Comment by marefr** [comment]
Please include full dashboard JSON if you think that's related.

**Comment by marefr** [comment]
@mtanda seems related to [this](https://github.com/grafana/grafana/pull/13571/files#diff-ed5cc68bef51197743c5863b078db25bR138) change, but the only way I think above can happen would be if there are multiple queries with the same `RefId`, right?

**Comment by mtanda** [comment]
@marefr I guess the same `RefId` is not a problem.
I should protect map by mutex. It's a common mistake about golang map...

**Comment by mtanda** [comment]
@marefr The line is derived from another PR.
It is released at 5.3 beta.

https://github.com/grafana/grafana/pull/11624/files#diff-ed5cc68bef51197743c5863b078db25bR124

https://github.com/grafana/grafana/pull/13684 will fix this issue, please check.
